src/Compilers/CSharp/Portable/Symbols/TypeWithState.cs

```diff
@@ -28,6 +29,44 @@ public static TypeWithState Create(TypeSymbol type, NullableFlowState defaultSta
             return new TypeWithState(type, state);
         }
 
+        public static TypeWithState Create(TypeWithAnnotations typeWithAnnotations, FlowAnalysisAnnotations annotations = FlowAnalysisAnnotations.None)
+        {
+            var type = typeWithAnnotations.Type;
+            if (type is null)
+            {
+                return new TypeWithState(null, NullableFlowState.MaybeNull);
+            }
+
+            NullableFlowState state;
+            if (type.CanContainNull())
+            {
+                if ((annotations & FlowAnalysisAnnotations.MaybeNull) != 0)
```

Comment:
Did LDM confirm that MaybeNull takes precedence over NotNull?

Reply:
This seems sensible (conservative/safe to return the worst case state).
I took a note in [https://github.com/dotnet/csharplang/issues/2201](https://github.com/dotnet/csharplang/issues/2201) to let LDM know.

---
In reply to: [288212991](https://github.com/dotnet/roslyn/pull/35955#discussion_r288212991) [](ancestors = 288212991)